src/pb_stub.cc

```diff
@@ -771,10 +771,17 @@ Stub::ProcessRequests(RequestBatch* request_batch_shm_ptr)
           std::to_string(response_size) + "\n";
       throw PythonBackendException(err);
     }
-    for (auto& response : responses) {
+    for (size_t i = 0; i < response_size; i++) {
+      // If the model has checked for cancellation and the request is cancelled,
+      // replace returned type with a cancelled response.
+      if (py_request_list[i].cast<InferRequest*>()->IsCancelledLastResponse()) {
+        responses[i] = std::make_shared<InferResponse>(
+            std::vector<std::shared_ptr<PbTensor>>{},
+            std::make_shared<PbError>("", TRITONSERVER_ERROR_CANCELLED));
+      }
       // Check the return type of execute function.
-      if (!py::isinstance<InferResponse>(response)) {
-        std::string str = py::str(response.get_type());
+      else if (!py::isinstance<InferResponse>(responses[i])) {
```

Comment:
Let's keep the logic here simple. We would return whatever response they have returned and shouldn't try to create cancelled requests automatically.

Reply:
Yes. Reverted the logic change: [Rollback non-decoupled any response on cancel](https://github.com/triton-inference-server/python_backend/pull/304/commits/3de5922c48ece05b5d55862063e67ab3a22c97f5)